packages/protocol/contracts/tokenvault/ERC20Vault.sol

```diff
@@ -229,11 +295,13 @@ contract ERC20Vault is BaseVault {
         // If it's a bridged token
         if (bridgedToCanonical[token].addr != address(0)) {
             ctoken = bridgedToCanonical[token];
-            IMintableERC20(token).burn(msg.sender, amount);
+            IBridgedERC20(token).burn(msg.sender, amount);
             _balanceChange = amount;
         } else {
+            if (btokenBlacklist[token]) revert VAULT_BTOKEN_BLACKLISTED();
```

Comment:
i think this check (`if (btokenBlacklist[token])`) shall go into the `if` clause not into the `else`? 🤔 
Because blacklisting only lives on non-canonical chain (bc. it is bridged). So basically we shall not allow it to be bridged back - but migrate first.

Reply:
Please take a look at[ this commit](https://github.com/taikoxyz/taiko-mono/pull/15255/commits/729e155044e3f81b4a8f21d87df34c2ab2644bc6). And do let me know if this is what you think is correct.

Reply:
Yep, here is correct ! 🖖